### PR TITLE
WIP : Add regtest mode

### PR DESCRIPTION
**Pull request by dannasessha**
WIP!

Following up on [an older issue](https://github.com/zingolabs/zingolib_original/issues/85).

Solves #6 with first commit. 

<strike>Currently a non-functional flag that is recognized when running the compiled binary (ex, `./target/release/zingo-cli --regtest`).</strike>

### CODE REVIEW

File: cli/src/version.rs
Comment (zancas) [inline cli/src/version.rs:1]: +1

**Comment by dannasessha** [comment]
Posting here a message I've received: 
"Why not pass the zcashd and lightwalletd binaries to --regtest as arguments? Force a user to figure out where to get those binaries from.. that's not a problem that you need to solve inside this feature."

Sounds reasonable to me. I could probably easily also include default values that fail if they don't find something.

**Comment by zancas** [comment]
Adding sane defaults, is beyond the scope of this task.   If the User doesn't provide binaries the `zingo-cli` should error out and exit.   The developer of this feature, @dannasessha shouldn't need to write any special logic to get this behavior.

**Comment by AloeareV** [comment]
`FakeMainnet` uses the mainnet parameters defined here: https://github.com/zcash/librustzcash/blob/main/zcash_primitives/src/constants/mainnet.rs, with its only difference being all the network upgrade heights set to 1. 
the regtest parameters are different, defined here: https://github.com/zcash/librustzcash/blob/main/zcash_primitives/src/constants/regtest.rs. For minimum viable setup we can set the network upgrade heights to 1, but eventually in theory we should be parsing the `zcash.conf` and parsing the network upgrade heights defined in there.

**Comment by AloeareV** [comment]
If we could get the 'press a button to boot a zcashd a, lightwalletd, and a zingo-cli that all connect to each other' for mainnet and/or testnet, that would be fantastic to get in, I don't want that functionality to be blocked on regtest.

**Comment by AloeareV** [comment]
The network enum is inspired by a similar enum in librustzcash defined here: https://github.com/zcash/librustzcash/blob/0142a3db1b16d3859a013a7079587ae7c5882afc/zcash_primitives/src/consensus.rs#L274-L278

**Comment by dannasessha** [comment]
Acknowledged, I think I understand, I'll look more  into it.

**Comment by dannasessha** [comment]
I'd like a review but I think I have got Network:Regtest working, tested locally.

README is attempting to be very verbose with the state of this branch also.

**Review by AloeareV — APPROVED** [review]
This looks good! Fixes #12. Will spawn a new issue of our regtest mode expecting specific NU heights, but I don't think being able to be flexible for custom network upgrade heights is high-priority. 

Zancas was specifically requested as a reviewer, which I interpret as a desire for zancas to see this before its merged in. I won't hit merge, and will wait and allow @zancas to merge this, or @dannasessha if you wish to merge before that happens!

**Review by skyl — APPROVED** [review]
LGTM

**Comment by skyl** [comment]
> The network enum is inspired by a similar enum in librustzcash defined here: https://github.com/zcash/librustzcash/blob/0142a3db1b16d3859a013a7079587ae7c5882afc/zcash_primitives/src/consensus.rs#L274-L278

Maybe it makes sense to eventually contribute the regtest upstream and depend on that definition?

**Comment by zancas** [comment]
This looks awesome!   It's still marked `WIP`, and labeled `DRAFT`.   I am +1 on merging this, if that's correct then that mark and label should change.

**Review by zancas — APPROVED** [review]
LGTM!  I can't remove `DRAFT` or `WIP` but this seems like a landable chunk of work, so I am merging!

**Pull request merged**
